k8s_util/k8s_util.go

```diff
@@ -340,24 +340,29 @@ func GetLog(podName, namespace string) (string, error) {
 	// We can't declare a variable somewhere which can be skipped by goto
 	var req *rest.Request
 	var readCloser io.ReadCloser
-	var err error
-	bytes := []byte{}
-	restClient, err := GetRESTClient()
+
+	buf := new(bytes.Buffer)
+	clientset, err := GetClientset()
 	if err != nil {
 		goto use_kubectl
 	}
 
-	req = restClient.Get().Namespace(namespace).Name(podName).Resource("pods").SubResource("log")
+	req = clientset.CoreV1().Pods(namespace).GetLogs(
+		podName,
+		&core_v1.PodLogOptions{},
+	)
+
 	readCloser, err = req.Stream()
+	defer readCloser.Close()
 	if err != nil {
 		goto use_kubectl
 	}
 
-	defer readCloser.Close()
-	_, err = readCloser.Read(bytes)
-	if err == nil {
-		return string(bytes), nil
+	buf.ReadFrom(readCloser)
+	if Debug {
+		fmt.Println("Log of Pod ", podName, " in Namespace ", namespace, " through API:\n", buf.String())
```

Comment:
Can you remove spaces from the strings here?

Reply:
And put last buf.String() in new fmt.Println. Remove \n from second last as well.

Reply:
Addressed. Thanks